lib/at_parser/at_match.re

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* Generated by re2c 3.0
```

Comment:
As someone unfamiliar with re2c, it's a bit confusing at first glance that the .re file has a "Generated by re2c" comment. Unless this form is standard practice, maybe we should say "at_match.c generated by re2c 3.0" to make it immediately clear that this file is the human-written source.